### PR TITLE
Atomic Hosting: Change path to /hosting-admin

### PR DESCRIPTION
**Pull request by mmtr**
#### Changes proposed in this Pull Request

Change the path used for the Atomic Hosting Dashboard to `/wp-hosting`.

We cannot use the `/hosting` path because https://wordpress.com/hosting is currently being used as a landing page.

#### Testing instructions

* If using a local install, be sure to do a fresh start so that the new routes can take effect (`npm start`).
* Switch to an Atomic site.
* Select "Manage > SFTP & MySQL" in the sidebar.
* Make sure it redirects to `/wp-hosting/:site`.
* Make sure the "Manage > SFTP & MySQL" sidebar item is marked as active.
* Go to `/wp-hosting`.
* Confirm the URL works and displays a site selector.
* Select an Atomic site.
* Verify the URL changes to `/wp-hosting/:site`.

### CODE REVIEW

**Comment by gwwar** [comment]
Summarizing some other options in a slack chat with Team Atomic:
- `/hosted`
- `/wp-hosting`
- `/hosting-admin`
- `/hosting-config`

**Comment by mmtr** [comment]
Thanks for the suggestions! I sticked with `/wp-hosting` since it sounded more semantic/natural. Happy to change it to another one if folks have other preferences.

**Comment by gwwar** [comment]
Would it be possible to get a final sign-off on the name here from @apeatling @bazza or @jmdodd ?

**Comment by kwight** [comment]
> I sticked with /wp-hosting since it sounds more semantic/natural to me.

I'll disagree! 🙃It's basically a prefix, that we don't use anywhere else in Calypso when naming routes (I don't think, at least). I'd prefer `hosting-admin` or hosting-access` ("what are we managing? access to our hosting..." ¯\_(ツ)_/¯)

**Comment by kwight** [comment]
Nice! A few things:

- I've added a note to the instructions about restarting Calypso for the new route.
- I noticed that directly loading the route with an invalid site redirects to the site root, rather than `/wp-hosting` to select a valid site (which I would expect).
- `/wp-hosting` should have filtered results, so that only valid sites can be selected.

**Comment by gwwar** [comment]
I'd prefer `hosted` since it's shorter, but we should get a confirmation from the Atomic before we land this.

**Comment by mmtr** [comment]
> I'd prefer `hosted` since it's shorter, but we should get a confirmation from the Atomic before we land this.

I like shorter URLs too, but I always prefer semantic URLs over short URLs if we cannot have both. `/hosted` sounds confusing to me as opposed to `/hosting-access` as @kwight suggested.

However, `/hosted` was suggested by Atomic (p1572461613241600-slack-atomic) so maybe it's just me struggling with that term 🙂

@apeatling do you have any thoughts?

**Comment by apeatling** [comment]
I don't like `/hosted` it doesn't mean anything and is confusing. I much prefer `/hosting-admin` because I read that and have a fairly good idea of what to expect on the page.

**Comment by gwwar** [comment]
From slack, @jmdodd says that `/hosting-admin` is fine

**Comment by gwwar** [comment]
> Could we start using constants for strings that are reused?

We normally do @aidvu, but it's not common for routing. For, me it's personally more readable to see `/my-route/:my-param` than a helper function using a constant. Let's defer on this one until Calypso updates project recommendations.

File: client/my-sites/sites/index.jsx [hunk 1]
```diff
@@ -93,6 +93,9 @@ class Sites extends Component {
 			case 'home':
 				path = translate( 'My Home' );
 				break;
+			case 'hosting-admin':
+				path = translate( 'SFTP & MySQL' );
```
Comment (gwwar) [inline client/my-sites/sites/index.jsx]: http://calypso.localhost:3000/hosting-admin if folks visit directly

Screen Shot 2019-11-04 at 1 14 14 PM

Not a blocker, but let's align the page title with what we pick for copy in the section.

Screen Shot 2019-11-04 at 1 15 14 PM

Comment (mmtr) [reply client/my-sites/sites/index.jsx]: https://github.com/Automattic/wp-calypso/pull/37321

**Pull request merged**